### PR TITLE
Use latest Swagger without using Current Swagger bundle

### PR DESCRIPTION
**Pull request by pchojnacki**
Its my intution that we would benefit long term from integration of most of current swagger bundle into pandora codebase. As that would make updating upstream Swagger code much easier.

Additionally this PR is first step in configuring Building both swagger specs and swagger generated source code during gradle build.
This will allow for compile time validation of interface compatibility.

This change was mostly inspired by missing support for File uploads in old swagger.
Old swagger bundle is still maintained as a means to provide seamless and gradual upgrade path for other services. Also inbetween version swagger changed their package names: https://github.com/swagger-api/swagger-core/pull/1128/files

Currently the biggest feature missing is integrated swagger ui, so right now its only for those services that are OK with that.

/cc: @nmonterroso @drsnyder @sqreek @harnash  @ArturKlajnerok

### CODE REVIEW

File: service/static-assets/src/main/java/com/wikia/services/staticassets/resources/StaticAssetsResource.java [hunk 1]
```diff
@@ -42,6 +44,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.util.Date;
```
Comment (WikiaTech) [inline service/static-assets/src/main/java/com/wikia/services/staticassets/resources/StaticAssetsResource.java]: MINOR Remove this unused import 'java.util.Date'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a private constructor to hide the implicit public one.](https://github.com/Wikia/pandora/blob/5c1188d4ea4f547315ff5ac142dc99599f91f1a6/lib/service-core/src/main/java/com/wikia/pandora/http/Util.java#L11) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

**Comment by nmonterroso** [comment]
Awesome, thanks for taking the time to set this up so other services can take advantage of it! As for source code generation we have the source code downloader thing in the central swagger ui, though maybe the UI for that could be improved and the dependency for that could be updated as well.

How do you envision source code generation working here? I'm curious since there are a lot of options depending on the language the code is being generated for.

**Comment by pchojnacki** [comment]
@nmonterroso Right now I just would like to handle simplest case of creating Java clients from Java code and being able to reuse them in Pandora.

**Comment by drsnyder** [comment]
:+1:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a private constructor to hide the implicit public one.](https://github.com/Wikia/pandora/blob/52eb340474785b84e5850417b7c4c70681e16d83/lib/service-core/src/main/java/com/wikia/pandora/http/Util.java#L11) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

File: lib/service-core/src/main/java/com/wikia/pandora/http/Util.java [hunk 2]
```diff
@@ -22,15 +22,19 @@
    *
    * @param environment
    */
+  @Deprecated
   public static void configurePublicCors(Environment environment) {
```
Comment (WikiaTech) [inline lib/service-core/src/main/java/com/wikia/pandora/http/Util.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: lib/service-core/src/main/java/com/wikia/pandora/http/Util.java [hunk 3]
```diff
@@ -22,15 +22,19 @@
...
+    configureCors(environment, "/*");
+  }
+  @Deprecated
+  public static void configureCors(Environment environment, String... urlPatterns) {
```
Comment (WikiaTech) [inline lib/service-core/src/main/java/com/wikia/pandora/http/Util.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues:
* MAJOR 5 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a private constructor to hide the implicit public one.](https://github.com/Wikia/pandora/blob/92bbf78d75d088293062e0e93c8bd51a048fe1cd/lib/service-core/src/main/java/com/wikia/pandora/http/Util.java#L11) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)
* MINOR [Remove this unused import 'com.wikia.pandora.core.swagger.SwaggerBundleProvider'.](https://github.com/Wikia/pandora/blob/92bbf78d75d088293062e0e93c8bd51a048fe1cd/service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java#L7) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)
* MAJOR [The Cyclomatic Complexity of this method "doUpdateOrCreateUserAvatar" is 11 which is greater than 10 authorized.](https://github.com/Wikia/pandora/blob/92bbf78d75d088293062e0e93c8bd51a048fe1cd/service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java#L203) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* MAJOR [Refactor this method to throw at most one checked exception instead of: java.sql.SQLException, java.net.URISyntaxException](https://github.com/Wikia/pandora/blob/92bbf78d75d088293062e0e93c8bd51a048fe1cd/service/user-preference/src/main/java/com/wikia/userpreference/resources/UserPreferenceResource.java#L74) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)

File: lib/service-core/src/main/java/com/wikia/pandora/http/Util.java (hunk 2 above)
Comment (WikiaTech) [inline lib/service-core/src/main/java/com/wikia/pandora/http/Util.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

File: lib/service-core/src/main/java/com/wikia/pandora/http/Util.java (hunk 3 above)
Comment (WikiaTech) [inline lib/service-core/src/main/java/com/wikia/pandora/http/Util.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 3 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a private constructor to hide the implicit public one.](https://github.com/Wikia/pandora/blob/4aefdff88eb9037dc9c84c9bc2f52943d4a73b9e/lib/service-core/src/main/java/com/wikia/pandora/http/Util.java#L11) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)
* MINOR [Remove this unused import 'com.wikia.pandora.core.swagger.SwaggerBundleProvider'.](https://github.com/Wikia/pandora/blob/4aefdff88eb9037dc9c84c9bc2f52943d4a73b9e/service/user-avatar/src/main/java/com/wikia/services/useravatar/UserAvatarApplication.java#L7) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by pchojnacki** [comment]
Swagger ui is still missing.
But I've finished generation of Java code from swagger annotated Java code.
Working examples are in static-assets/generated-client and user-avatar/generated-client.

I wasn't integrating that generated code anywhere Yet. But although currently the generated code compiles , I think some tweaks to generation schema might be in order before that is achieved.

/cc: please review @Wikia/services-team

**Comment by nmonterroso** [comment]
I think it's OK (and maybe even preferable?) for the Swagger UI to be missing per service, since we have the centralized UI.

File: lib/service-core/src/main/java/com/wikia/pandora/http/Util.java [hunk 4]
```diff
@@ -10,27 +10,32 @@
...
-   *    ...
-   *    configurePublicCors(environment);
+   * public void run(UserPreferenceConfiguration configuration, Environment environment) ...
+   * configurePublicCors(environment);
    *
-   * @param environment
+   * @deprecated once all uses are refactored out this should be removed
    */
+  @Deprecated
   public static void configurePublicCors(Environment environment) {
```
Comment (WikiaTech) [inline lib/service-core/src/main/java/com/wikia/pandora/http/Util.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by pchojnacki** [comment]
I think people would still like to use swagger-ui with local data source. But that could be provided via other means. Like Central swagger Ui querying local swagger.json or some local single page app.

But yeah I think it would  be best not to bundle swagger-ui with every application runtime.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by nmonterroso** [comment]
I'm working on something that would benefit from this, but it's not too clear how to get it all set up. Can you write some docs in either a `README` or on [the wiki](https://github.com/Wikia/pandora/wiki)?

Other than that LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by nmonterroso** [comment]
LGTM

**Pull request merged**